src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -144,7 +144,19 @@ def run(
                 return BatchResult.create(
                     self._start_time, datetime.utcnow(), [], AggregationResult({}, {}, {}), status=Status.Canceled
                 )
-            raise e
+            elif isinstance(e, PromptflowException):
+                raise e
+            else:
+                # For unexpected error, we need to wrap it to SystemErrorException.
+                # This allows us to see the stack trace inside.
+                unexpected_error = UnexpectedError(
+                    target=ErrorTarget.BATCH,
+                    message_format=(
+                        "Unexpected error occurred while executing the batch run. The error details: {error_message}."
```

Comment:
Shall we also expose the inner exception type?
Like: "Unexpected error occurred while executing the batch run: {error_type_and_message}"
![image](https://github.com/microsoft/promptflow/assets/113209/ec66b698-fed8-415b-a5df-f2b4678d6aa8)

Reply:
Updated.